### PR TITLE
extracted class - argumentsParser

### PR DESCRIPTION
**Pull request by carloscaldas**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/1829)

### CODE REVIEW

**Comment by carloscaldas** [comment]
test!

File: src/main/scala/beam/sim/ArgumentsParser.scala [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          else success
+      )
+    }
+  }
+
+  def parseArguments(parser: OptionParser[Arguments], args: Array[String]): Option[Arguments] = {
+    parser.parse(args, init = Arguments())
+  }
+
+  def parseArguments(args: Array[String]): Option[Arguments] = parseArguments(buildParser, args)
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/ArgumentsParser.scala]: This seems to be the only public method. Maybe we should hide the rest unless it becomes necessary to expose it?

Comment (carloscaldas) [reply src/main/scala/beam/sim/ArgumentsParser.scala]: done

**Comment by carloscaldas** [comment]
test!

**Pull request merged**